File: js/packages/app/src/__tests__/features/object-references/external/polywrap/object-references-example/src/polywrap/external/module/ImportBindings.ts
import { TestObjectGetterImport } from "../classes/TestObjectGetter";
import { TestObjectGetter } from "../classes";


import { TestExternalClass } from "../../..";



export class ImportBindings {
  constructor(
    
    public testReceiveReference: (
      arg: TestExternalClass,
    ) => Promise<string>,
        
    public testInvokeExternalGlobalFunction: (
      arg: string,
    ) => Promise<string>,
        
    public testInvokeExternalStaticMethod: (
      arg: string,
    ) => Promise<string>,
        
    public testInvokeExternalInstanceMethod: (
      arg: string,
    ) => Promise<string>,
            
    
    public TestObjectGetter: TestObjectGetterImport,
            
  ) {
  }
}

File: rust/app/src/polywrap/external_references/src/polywrap/external/module/ImportBindings.rs
use external::classes::TestObjectGetter::{ TestObjectGetterImport };
use external::{ TestObjectGetter };


use create::{ TestExternalClass };



struct ImportBindings {
    
    pub testReceiveReference: Fn(
      TestExternalClass,
    ) -> string,
        
    pub testInvokeExternalGlobalFunction: Fn(
      string,
    ) -> string,
        
    pub testInvokeExternalStaticMethod: Fn(
      string,
    ) -> string,
        
    pub testInvokeExternalInstanceMethod: Fn(
      string,
    ) -> string,
            
    
    pub TestObjectGetter: TestObjectGetterImport,
            
}

impl ImportBindings {
    pub fn new(
        
        testReceiveReference: (
        arg: TestExternalClass,
        ) -> string,
                
        testInvokeExternalGlobalFunction: (
        arg: string,
        ) -> string,
                
        testInvokeExternalStaticMethod: (
        arg: string,
        ) -> string,
                
        testInvokeExternalInstanceMethod: (
        arg: string,
        ) -> string,
                        
        
        TestObjectGetter: TestObjectGetterImport,
                        
    ) -> Self {
        Self {
            
            testReceiveReference: testReceiveReference,
                        
            testInvokeExternalGlobalFunction: testInvokeExternalGlobalFunction,
                        
            testInvokeExternalStaticMethod: testInvokeExternalStaticMethod,
                        
            testInvokeExternalInstanceMethod: testInvokeExternalInstanceMethod,
                                    
            
            TestObjectGetter: TestObjectGetter,
                                    
        }
    }
}
